### ISSUE TITLE
Unstable Hlink speeds

### ISSUE THREAD

**Issue by Sheldonsuckz** [Kyle-Kyle/baidudl#158]
Since yesterday speeds from generated Hlinks have been fluctuating greatly. For example the download will start at around 2MB/s but then start to dip down to the 100 kb/s range.  Then the speed will gradually increase again until it hets to about 300kb/s and then again fall to 100kb/s.
What i seemed to notice is the speed is directly related to the number of connections in the downloader. It starts off at about CN:17 and then by the end we have a CN:3. Another thing I find weird is that some connections are extremely stable and I can get a CN:100 with amazing speeds.

This picture is of a file that started out fast but then lost connections.
<img width="575" alt="screen shot 2018-11-09 at 11 03 21 am" src="https://user-images.githubusercontent.com/35752172/48273525-25864900-e40f-11e8-8367-b856ab2e04f5.png">

While this picture is of a file that had a stable CN:100 count throughout the download.
<img width="563" alt="screen shot 2018-11-08 at 2 40 22 pm" src="https://user-images.githubusercontent.com/35752172/48273582-4353ae00-e40f-11e8-8647-83809240b71e.png">

**Comment by Kyle-Kyle** [comment]
a very important report.
Were you using the same baidu account? I could be an account issue.

**Comment by Sheldonsuckz** [comment]
Yes I was using the same account. It seems I can not get very fast speeds anymore. All files I tried today give extremely slow speeds.

**Comment by xayoung** [comment]
same.

**Comment by Sheldonsuckz** [comment]
Got an error today of  [NameResolver.cc:60] errorCode=19 Failed to resolve the hostname qdall01.baidupcs.com, cause: nodename nor servname provided, or not known.
Seems Baidu has figured out which of their addresses were giving unwanted speeds.

**Comment by Sheldonsuckz** [comment]
I think it has officially been patched today. Now no matter how many connections I get the speed is slow. Maybe it is account related but speeds like this are even slower than the official netdisk app.
<img width="624" alt="screen shot 2018-11-12 at 10 17 57 am" src="https://user-images.githubusercontent.com/35752172/48356617-a68a4e00-e664-11e8-9be4-644bb927e237.png">

**Comment by citterino** [comment]
@Sheldonsuckz Have you been using the official netdisk app when baidudl doesn't work? The main reason I refuse to use Baidu's netdisk app is because it may be a spyware, I mean who knows but it's better to be on a safer side. What's your thought on this?

**Comment by Sheldonsuckz** [comment]
I use the netdisk app only as a last resort to be honest. 
While I don't really care if it's spyware the speed on the netdisk app fluctuates and the downloads occasionally fail. I still use baidudl to download at the moment for files larger than 300MB. 
For files under 300MB there is a method that allows for 2MB/s download but I'm not sure if I am allowed to mention it on this github. If you search google you'll find it.

**Comment by Sheldonsuckz** [comment]
Just had to make sure.
https://github.com/gxvv/ex-baiduyunpan
This script allows for files under 300MB to be downloaded at decent speed. Over 300MB you will get a "400 bad request" error.

**Comment by Sheldonsuckz** [comment]
<img width="592" alt="screen shot 2018-11-16 at 5 46 39 pm" src="https://user-images.githubusercontent.com/35752172/48651225-d8364880-e9c7-11e8-8e17-5d3b8edc9ab0.png">
<img width="591" alt="screen shot 2018-11-16 at 5 45 44 pm" src="https://user-images.githubusercontent.com/35752172/48651226-d8364880-e9c7-11e8-8718-221176bdbece.png">
Very interesting but today I saw the console load with a speed of 2980. i have never seen anything over 80, And when that loaded I once again was able to get extremely fast speeds. At one point it was downloading at 14 MB/s.

**Comment by Sheldonsuckz** [comment]
To be honest I'm not sure. I have a couple of friends who also can't generate the Hlinks. The only difference is that I am on OSX and they are on Windows.

**Comment by rkaas10** [comment]
> To be honest I'm not sure. I have a couple of friends who also can't generate the Hlinks. The only difference is that I am on OSX and they are on Windows.

And you are using Chrome or an another browser?

**Comment by Sheldonsuckz** [comment]
Google Chrome Version 70.0.3538.102 (Official Build) (64-bit)
Hlinks aren't generating fast speeds at the moment so it is comparable to just use the netdisk app.

**Comment by rkaas10** [comment]
@Sheldonsuckz Thanks for answering my questions. I really appreciate it.

Actually, I'm very curious what speed you now get with your Hlinks.

**Comment by Sheldonsuckz** [comment]
Really depends on the day.
Yesterday I was getting 9-10 MB/s but today I'm back down to 50 kB/s.

**Comment by Sheldonsuckz** [comment]
I'm in New York I really don't understand how it could be so different.

**Comment by Kyle-Kyle** [comment]
@Sheldonsuckz Because CDN. And it depends on your physical location.

**Comment by Kyle-Kyle** [comment]
@rkaas10 
You don't want to download with HLink or GLink. I may disable them later.
You need to use RPCDownload.
If you can get 300k with only HLink or GLink, you can get 3M at least with RPCDownload

**Comment by Kyle-Kyle** [comment]
Interesting, I read your chat. It looks that you cannot generate links on windows? I'm going to test it.

**Comment by Sheldonsuckz** [comment]
I'll try with a bunch of different IPs later and post the results. At first I thought it may had to have been the number of accounts logged into a certain IP at the same time. But, even at my home the speed is still slow.

**Comment by Sheldonsuckz** [comment]
Just tried again with my own IP and I got a Hlink that gave me 0bps
<img width="739" alt="screen shot 2018-11-17 at 1 57 51 pm" src="https://user-images.githubusercontent.com/35752172/48664725-06239780-ea71-11e8-9740-2b0fcb3ba1d5.png">
<img width="1279" alt="screen shot 2018-11-17 at 1 58 36 pm" src="https://user-images.githubusercontent.com/35752172/48664727-0cb20f00-ea71-11e8-8477-c5909853c9b5.png">

**Comment by Kyle-Kyle** [comment]
@Sheldonsuckz Can you wait for a bit longer? It may continue download later.

**Comment by Sheldonsuckz** [comment]
<img width="1287" alt="screen shot 2018-11-17 at 2 02 02 pm" src="https://user-images.githubusercontent.com/35752172/48664752-66b2d480-ea71-11e8-9704-d95321d4c7de.png">
That is the fastest speed I got today. And this is the same file that I just posted.

**Comment by Kyle-Kyle** [comment]
@Sheldonsuckz can you share the file with me and let me try it? Now sure it is an account issue or location issue.

**Comment by Sheldonsuckz** [comment]
Sure I'll email you the file.

**Comment by Kyle-Kyle** [comment]
@Sheldonsuckz Please send me the link of the file instead of the file. Thanks.

**Comment by Sheldonsuckz** [comment]
链接: https://pan.baidu.com/s/1Nfskp1B70SeimRGwEIYQQA 提取码: zjpt  

Tell me after you're done so I can unshare this

**Comment by Kyle-Kyle** [comment]
@Sheldonsuckz After tried your file. I cannot get download speed on other files. lol

**Comment by Sheldonsuckz** [comment]
Very interesting

**Comment by Kyle-Kyle** [comment]
@Sheldonsuckz I suspect this is an account issue. After the account get banned, you have to suffer the slow download speed for at least a week.

**Comment by Kyle-Kyle** [comment]
@Sheldonsuckz But the question is, I know there is a way to bypass that, but don't know how.

**Comment by Sheldonsuckz** [comment]
Hmm okay well i wouldn't be surprised if my accounts are banned. I have been using these accelerators for a while.  Just kinda weird that it happened exactly as the scripts stopped working. And just to confirm, have you been able to generate anymore high speed links since you tried my file? 
Thank you for your help btw, I was probably just gonna buy a SVIP next week if all else fails.

**Comment by Kyle-Kyle** [comment]
@Sheldonsuckz not everything is not working. `pandownload` is working. You can get 8M/s sometimes. most of the time it can give me 1M/s

**Comment by Sheldonsuckz** [comment]
is there a OSX version on pandownload? Or do I have to run it through a virtual machine?

**Comment by Kyle-Kyle** [comment]
@Sheldonsuckz I'm afraid you have to run it through a vm

**Comment by rezkysaragih** [comment]
@Kyle-Kyle Actually, I always get error results, while try to get HLinks and also RPCDownload (In Windows). @Sheldonsuckz said he can download from OSX. Idk, but today i can generate HLinks with RPCDownload too (wooaahhh) but the speeds just same with script in greasyfork
![cats](https://user-images.githubusercontent.com/26543996/48668072-1b530180-eb17-11e8-85c3-2092f1e9f87b.jpg)

**Comment by Kyle-Kyle** [comment]
@rezkysaragih the link generated by greasyfork is the same to GLink. This extension is supposed to be better than that but apparently it is not now. I'll work on it to make it great again.

**Comment by Sheldonsuckz** [comment]
Maybe it is more account related than i thought.
<img width="401" alt="screen shot 2018-11-17 at 11 38 27 pm" src="https://user-images.githubusercontent.com/35752172/48668691-fe401380-eac1-11e8-81d2-942f40e32d2e.png">
Even Pandownload gives me very slow speeds

**Comment by kawashiro310** [comment]
@Sheldonsuckz Your account may have been speed limited.

**Comment by rezkysaragih** [comment]
> @rezkysaragih the link generated by greasyfork is the same to GLink. This extension is supposed to be better than that but apparently it is not now. I'll work on it to make it great again.

Ah... . I Guess will wait your your update vers. project. Good luck 😉 😉 😂

**Comment by rezkysaragih** [comment]
> @Kyle-Kyle  the link generated by greasyfork is the same to GLink. This extension is supposed to be better than that but apparently it is not now. I'll work on it to make it great again.

Ah... . I Guess will wait for your update vers. project. Good luck 😉 😉 😂

**Comment by Sheldonsuckz** [comment]
@Kyle-Kyle So i ended up making another baidu account and found that PanDownload works great as it maxes out my connection. If I were to download using your extension, would that count towards my limit. Since it appears the limit of downloading is quite small.

**Comment by darkred3** [comment]
@Sheldonsuckz 
Tried pandownload as well but it only ever downloads for 2-4seconds before disconnecting and reconnecting again. Do you perhaps know anything about that?

**Comment by Sheldonsuckz** [comment]
When you connect does ait have a red warning symbol? Also all 4 of my baidu accounts are banned in my apartment building. But when I'm on my university Internet all 4 of my accounts can download at 20MB/s. How many connections can you get ?.

**Comment by darkred3** [comment]
@Sheldonsuckz 
Yes it does. It seems like that means that my account may have been speed limited by baidu, but on what criteria does baidu do that?
I can get up to 4 connections but it really doesn't matter whether I set it to 100 connections or just 1, the download breaks down after a few seconds and resumes after a couple more.

**Comment by Sheldonsuckz** [comment]
Btw if you file is below 300MB hit the second button when you download and it will download decently fast even if your account is limited

**Comment by darkred3** [comment]
@Sheldonsuckz 
But if your problem was fixed by using a different computer at a different location(other IP) shouldn't it be IP related rather than account? Or did you use a different account as well?

**Comment by Sheldonsuckz** [comment]
I really can't say. Honestly I've tried so many things I don't know ow what works anymore. I'll be going home later in the week and will try from a fresh IP

**Comment by darkred3** [comment]
@Sheldonsuckz 
Just created a new account with an american phone number(like the other account) and got fastest speed my connection can get using pandownload. Used the same PC and same IP address.
I don't understand anything anymore...

**Comment by Sheldonsuckz** [comment]
Yes I did the same thing yesterday and then today my new account was limited and I only downloaded one file.

**Comment by darkred3** [comment]
@Sheldonsuckz 
Less than one day later the new account also seems to be speed limited. Though that might also have to do with it having been compromised and someone from a different continent logging in and downloading like 50GB ^^
